iceoryx_hoofs/source/posix_wrapper/named_pipe.cpp

```diff
@@ -94,15 +96,48 @@ NamedPipe::NamedPipe(const IpcChannelName_t& name,
 
     if (sharedMemory.has_error())
     {
-        std::cerr << "Unable to open shared memory: \"" << convertName(name) << "\" for named pipe \"" << name << "\""
-                  << std::endl;
+        std::cerr << "Unable to open shared memory: \"" << convertName(NAMED_PIPE_PREFIX, name)
+                  << "\" for named pipe \"" << name << "\"" << std::endl;
         m_isInitialized = false;
         m_errorValue = (channelSide == IpcChannelSide::CLIENT) ? IpcChannelError::NO_SUCH_CHANNEL
                                                                : IpcChannelError::INTERNAL_LOGIC_ERROR;
         return;
     }
 
-    m_sharedMemory.emplace(std::move(sharedMemory.value()));
+    cxx::expected<Semaphore, SemaphoreError> sendSemaphore = cxx::error<SemaphoreError>(SemaphoreError::UNDEFINED);
+    cxx::expected<Semaphore, SemaphoreError> receiveSemaphore = cxx::error<SemaphoreError>(SemaphoreError::UNDEFINED);
```

Comment:
It's kind of weird to have an `cxx::expected` as local variable and I would also discourage such habits. Can you please use a `cxx::optional` instead? Maybe we also need something similar to [`ok`](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok) fo the `expexted`

Reply:
completely rewritten.